docs/sql/snapshot_restore.txt

```diff
@@ -0,0 +1,326 @@
+.. highlight:: psql
+.. _snapshot-restore:
+
+Backup with Snapshot and Restore
+================================
+
+Snapshot
+--------
+
+In CRATE, backups are called *Snapshots*. They represent the state of
+the tables in a CRATE cluster at the time the *Snapshot* was
+created. A *Snapshot* is always stored in a *Repository* which has to
+be created up front.
+
+Creating a Repository
+~~~~~~~~~~~~~~~~~~~~~
+
+Repositories are used to store, manage and restore snapshots.
+
+They are created using the :ref:`ref-create-repository` statement::
+
+    cr> CREATE REPOSITORY where_my_snapshots_go TYPE fs
+    ... WITH (location='repo_path', compress=true);
+    CREATE OK (... sec)
+
+Repositories are uniquely identified by their name. Every repository
+has a specific type which determines how snapshots are stored.
+
+CRATE supports different repository types:
+:ref:`ref-create-repository-types-fs`, :ref:`ref-create-repository-types-hdfs`,
+and :ref:`ref-create-repository-types-url`. Support for further types
+might be added using :ref:`plugins`.
+
+The creation of a repository simply configures it inside the crate
+cluster. In general no data is written, no snapshots inside
+repositories are changed or deleted. This way you can tell the CRATE
+cluster about repositories that already contains snapshots.
+
+Creating a repository with the same name will result in an error::
+
+    cr> CREATE REPOSITORY where_my_snapshots_go TYPE fs
+    ... WITH (location='another_repo_path', compress=false);
+    SQLActionException[Repository 'where_my_snapshots_go' already exists]
+
+Creating a Snapshot
+~~~~~~~~~~~~~~~~~~~
+
+Snapshots are created inside a repository and can contain any number
+of tables. The :ref:`ref-create-snapshot` statement is used to create
+a snapshots::
+
+    cr> CREATE SNAPSHOT where_my_snapshots_go.snapshot1 ALL
+    ... WITH (wait_for_completion=true, ignore_unavailable=true);
+    CREATE OK (... sec)
+
+A snapshot is referenced by the name of the repository and the
+snapshot name, separated by a dot. If used with ``ALL``, all tables of
+the cluster (except :ref:`sys-tables <system-information>`,
+:ref:`information_schema` tables and blob tables) along with all
+persistent settings and the full cluster metadata are stored inside
+the snapshot.
+
+It is possible to select any number of tables for inclusion in the
+snapshot. In this case only the necessary metadata for those tables is
+included in the snapshot and the persistent settings will be excluded.
```

Comment:
is this metadata thingy relevant to the user? what are persistent settings in this regard?